### PR TITLE
Comments: Speeding up dashboard and comment moderation SQL load

### PR DESCRIPTION
**Pull request by uday-kokitkar**
Trac ticket: https://core.trac.wordpress.org/ticket/19901

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/comment/getCommentCount.php [hunk 1]
```diff
@@ -97,4 +98,40 @@ public function test_get_comment_count_post_trashed() {
 		$this->assertSame( 1, $count['post-trashed'] );
 		$this->assertSame( 0, $count['total_comments'] );
 	}
+
+	public function test_get_comment_count_validate_cache() {
```
Comment (spacedmonkey) [inline tests/phpunit/tests/comment/getCommentCount.php]: Please ensure that all new tests, include a `@ticket` number. All new tests, should also include a `@covers` as well.

File: tests/phpunit/tests/comment/getCommentCount.php [hunk 2]
```diff
@@ -97,4 +98,40 @@ public function test_get_comment_count_post_trashed() {
...
+		$count = get_comment_count( $post_id );
+
+		$this->assertSame( 1, $count['total_comments'] );
+
+		wp_delete_post( $post_id, true );
+
+		$count = get_comment_count( $post_id );
+
+		$this->assertSame( 0, $count['total_comments'] );
+	}
```
Comment (spacedmonkey) [inline tests/phpunit/tests/comment/getCommentCount.php:146]: I think we should add a test, where a comment is deleted and one where a comment change status.

Comment (uday-kokitkar) [reply tests/phpunit/tests/comment/getCommentCount.php:146]: Done. Added one more test.

File: tests/phpunit/tests/comment/getCommentCount.php (hunk 1 above)
Comment (uday-kokitkar) [reply tests/phpunit/tests/comment/getCommentCount.php]: Done.

File: tests/phpunit/tests/comment/getCommentCount.php [hunk 3]
```diff
@@ -97,4 +98,73 @@ public function test_get_comment_count_post_trashed() {
 		$this->assertSame( 1, $count['post-trashed'] );
 		$this->assertSame( 0, $count['total_comments'] );
 	}
+
+	/**
+	 * @ticket 19901
+	 *
+	 * @covers get_comment_count
```
Comment (spacedmonkey) [inline tests/phpunit/tests/comment/getCommentCount.php]:
```suggestion
	 * @covers ::get_comment_count
```

File: tests/phpunit/tests/comment/getCommentCount.php [hunk 4]
```diff
@@ -97,4 +98,73 @@ public function test_get_comment_count_post_trashed() {
...
+
+		$count = get_comment_count();
+
+		$this->assertSame( 0, $count['total_comments'] );
+	}
+
+	/**
+	 * @ticket 19901
+	 *
+	 * @covers get_comment_count
```
Comment (spacedmonkey) [inline tests/phpunit/tests/comment/getCommentCount.php]:
```suggestion
	 * @covers ::get_comment_count
```

File: tests/phpunit/tests/comment/getCommentCount.php [hunk 5]
```diff
@@ -97,4 +98,73 @@ public function test_get_comment_count_post_trashed() {
...
+
+		$count = get_comment_count( $post_id );
+
+		$this->assertSame( 0, $count['total_comments'] );
+	}
+
+	/**
+	 * @ticket 19901
+	 *
+	 * @covers get_comment_count
```
Comment (spacedmonkey) [inline tests/phpunit/tests/comment/getCommentCount.php]:
```suggestion
	 * @covers ::get_comment_count
```

**Comment by spacedmonkey** [comment]
CC @markjaquith for review.

**Review by peterwilsoncc — APPROVED** [review]
This looks good and manual tests work too.

I think there's a little room for the existing tests to be improved but that can be done on the generic test improvement ticket. 

@spacedmonkey this is good for commit.

**Pull request closed without merging**